server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java

```diff
@@ -607,18 +607,18 @@ public static ServerAddress startTServer(ThriftServerType serverType, TimedProce
             serverAddress = createBlockingServer(address, processor, protocolFactory,
                 maxMessageSize, serverName, numThreads, numSTThreads, timeBetweenThreadChecks);
             break;
-          case CUSTOM_HS_HA:
-            log.debug("Instantiating unsecure custom half-async Thrift server");
-            serverAddress = createNonBlockingServer(address, processor, protocolFactory, serverName,
-                numThreads, numSTThreads, timeBetweenThreadChecks, maxMessageSize);
-            break;
-          case THREADED_SELECTOR: // Intentional passthrough -- Our custom wrapper around threaded
-                                  // selector is the default
-          default:
+          case THREADED_SELECTOR:
             log.debug("Instantiating default, unsecure Threaded selector Thrift server");
             serverAddress = createThreadedSelectorServer(address, processor, protocolFactory,
                 serverName, numThreads, numSTThreads, timeBetweenThreadChecks, maxMessageSize);
             break;
+          case CUSTOM_HS_HA: // Intentional passthrough -- Our custom wrapper around threaded
+                             // selector is the default
+          default:
```

Comment:
default should fail here, instead of relying on this fall-through. We should ensure the default is set to one of the known types before we get here by setting the value to `ThriftServerType.getDefault()` if it is blank. If an unknown type is specified, we should simply fail... since we cannot recognize and satisfy the user's chosen configuration.

This buys us 3 things:

1. We don't have to re-arrange the switch statement whenever we change the default,
2. We can specify the default in a single place, and
3. We don't do auto-magic behavior, and instead fail sanely when we don't recognize the user's configuration.

Reply:
That's a subtle change to the behavior of an existing property, 2.0.0 seems like a good time to do something like that.

Reply:
I might argue it's a subtle "fix" in behavior... :smiley_cat: